genrl/deep/agents/vpg/vpg_temp.py

```diff
@@ -0,0 +1,195 @@
+from typing import Any, Dict, Optional, Tuple, Union
+
+import gym
+import numpy as np
+import torch
+import torch.optim as opt
+from torch.autograd import Variable
+
+from ....environments import VecEnv
+from ...common import (
+    BaseOnPolicyAgent,
+    RolloutBuffer,
+    get_env_properties,
+    get_model,
+    load_params,
+    save_params,
+    set_seeds,
+)
+
+
+class VPG(BaseOnPolicyAgent):
+    def __init__(
+        self,
+        network_type: str,
+        env: Union[gym.Env, VecEnv],
+        timesteps_per_actorbatch: int = 1000,
+        gamma: float = 0.99,
+        actor_batch_size: int = 4,
+        epochs: int = 1000,
+        lr_policy: float = 0.01,
+        lr_value: float = 0.0005,
+        policy_copy_interval: int = 20,
+        layers: Tuple = (32, 32),
+        seed: Optional[int] = None,
+        render: bool = False,
+        device: Union[torch.device, str] = "cpu",
+        run_num: int = None,
+        save_model: str = None,
+        load_model: str = None,
+        save_interval: int = 50,
+        rollout_size: int = 2048,
+    ):
+
+        super(VPG, self).__init__(
+            network_type,
+            env,
+            timesteps_per_actorbatch,
+            layers,
+            gamma,
+            lr_policy,
+            lr_value,
+            actor_batch_size,
+            epochs,
+            seed,
+            render,
+            device,
+            run_num,
+            save_model,
+            load_model,
+            save_interval,
+            rollout_size,
+        )
+        self.policy_copy_interval = policy_copy_interval
+
+        self.create_model()
```

Comment:
We should a bool argument like `setup_model` per basis we choose to call this method, otherwise its calling this every time

Reply:
Yeah, so I was saying for every time the agent is initialized, it is called. Maybe we can do it in another issue, nvm.